src/App.tsx

```diff
@@ -17,18 +43,34 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                setQuery={setQuery}
+                status={status}
+                setStatus={setStatus}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loading && <Loader />}
+              {!loading && filteredTodos.length > 0 && (
+                <TodoList
+                  selectedTodo={selectedTodo}
+                  todos={filteredTodos}
+                  onTodoSelect={todo => setSelectedTodo(todo)}
+                />
+              )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {selectedTodo && (
+        <TodoModal
+          todo={selectedTodo}
+          onCloseTodo={todo => setSelectedTodo(todo)}
```

Comment:
The `onCloseTodo` prop is expected to be a function that closes the modal. However, the current implementation passes a function that sets the selected todo, which might not be the intended behavior. Consider revising this function to properly handle the closing of the modal.